pandas/core/reshape/merge.py

```diff
@@ -1139,6 +1155,8 @@ def _get_merge_keys(self):
         # ugh, spaghetti re #733
         if _any(self.left_on) and _any(self.right_on):
             for lk, rk in zip(self.left_on, self.right_on):
+                deprecate_unknown_arraylike(lk)
```

Comment:
Might be a nicer user experience if only 1 warning was raised if `lk` and `rk` were both non-supported array-likes

Reply:
isnt that the default behavior of warnings?

Reply:
Ah right.